imessage/mac-nosip/nosip.go

```diff
@@ -87,6 +91,30 @@ func (mac *MacNoSIPConnector) Start() error {
 	}
 	mac.log.Debugln("Process started, PID", mac.proc.Process.Pid)
 	ipcProc.SetHandler(IncomingLog, mac.handleIncomingLog)
+	ipcProc.SetHandler(IncomingPong, mac.handleIncomingPong)
+
+	mac.killPinger()
+	pinger := time.NewTicker(time.Duration(15000) * time.Millisecond)
+	go func() {
+		// set to true initially to start the circuit
+		mac.receivedPing = true
+		for {
+			select {
+			case <-pinger.C:
+				// mac did not pong within 15s of last ping
+				if !mac.receivedPing {
+					mac.log.Fatalln("Barcelona did not respond to our last ping. Gotta go!")
+					os.Exit(-1)
```

Comment:
This is a weird way to write 255 but I guess it works if that's what you intended?